### PR TITLE
e2e: set --extra-create-metadata=false

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #

set "-extra-create-metadata=false" on
"create a Block mode PVC-PVC clone and bind it to an app"
to avoid ErrImageNotFound when rbd images are undergoing flattening
operation.

WorkAround: #2327
Signed-off-by: Prasanna Kumar Kalever \<prasanna.kalever@redhat.com\>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@pkalever this should be part of https://github.com/ceph/ceph-csi/pull/2898 and it's not fixing the problem of #2327 rather it just avoiding it.

**Review by nixpanic — CHANGES_REQUESTED** [review]
Instead of reading the yaml files from the `deploy/` directory, extend the functionality of `api/` and use that.

**Comment by pkalever** [comment]
> Instead of reading the yaml files from the `deploy/` directory, extend the functionality of `api/` and use that.

Where did you see that we are reading from `deploy/` directory? And can you explain a bit on which api you are referent to extend here? (may be elaborate a bit more please)

File: e2e/rbd.go [hunk 1]
```diff
@@ -2026,6 +2028,41 @@ var _ = Describe("RBD", func() {
 				if err != nil {
 					e2elog.Failf("failed to get server version: %v", err)
 				}
+
+				deploymentPatch, err := json.Marshal(map[string]interface{}{
```
Comment (nixpanic) [inline e2e/rbd.go:2032]: Why use a patch, and not just `Get()` and `Update()`. There should not be a need to list all `args` that way.

Comment (pkalever) [reply e2e/rbd.go:2032]: Because we need this for both helm and non-helm deployments.

Comment (nixpanic) [reply e2e/rbd.go:2032]: You can just use the Kubernetes API to get the Deployment object, modify the `args` and send the updated object to the API server. No need for JSON and untyped maps of strings.

Comment (pkalever) [reply e2e/rbd.go:2032]: Sure, I can give it a shot, thanks for the hint

**Comment by pkalever** [comment]
@Madhu-1 The tests are failing even with this workaround:

```
Apr  7 12:00:18.848: FAIL: backend images not matching kubernetes resource count,image count 2 kubernetes resource count 1
```

File: charts/ceph-csi-rbd/values.yaml [hunk 2]
```diff
@@ -128,13 +128,7 @@ provisioner:
...
   replicaCount: 3
   strategy:
-    # RollingUpdate strategy replaces old pods with new ones gradually,
-    # without incurring downtime.
-    type: RollingUpdate
-    rollingUpdate:
-      # maxUnavailable is the maximum number of pods that can be
-      # unavailable during the update process.
-      maxUnavailable: 50%
+    type: Recreate
```
Comment (nixpanic) [inline charts/ceph-csi-rbd/values.yaml:131]: What is the reason to change this? The comments from commit ea088d40be are very descriptive, not sure this is an improvement?

Comment (pkalever) [reply charts/ceph-csi-rbd/values.yaml:131]: Oh! sorry, this is just a workaround. Currently, when we set Strategy type to Rollingupdate, the Patch command is working differently, i.e while the old pod is still Running a new pod is getting created and waiting in the pending state, the old pod is not getting terminated (unless someone manually kill it) and the new pod is waiting for old one to get terminated, obviously, the status of pod replica is set to 2 out of which 1 is not ready. This scenario is leading to other failures in the CI, such as waiting for provisioner 1/2 state.

That said, when the `Rollingupdate: {}` attribute is set initially k8s won't give us the provision to patch/apply it to `null`, only manual edit is allowed.

That all said, this is just an experiment PR for now. Which is anyway not working as expected.

Comment (nixpanic) [reply charts/ceph-csi-rbd/values.yaml:131]: > That all said, this is just an experiment PR for now. Which is anyway not working as expected.

Ok! In that case, I'll mark this as a **Draft** for the time being.

**Comment by pkalever** [comment]
@Madhu-1 I hope we are good to close this PR then?

**Comment by Madhu-1** [comment]
> @Madhu-1 I hope we are good to close this PR then?

Yes we can close this one for now.

**Pull request closed without merging**